derivation.nix

```diff
@@ -112,18 +169,21 @@ in
                 inherit nativeBuildInputs;
 
                 unpackPhase = concatStringsSep "\n"
-                  (
+                  ( [ # components is an empty list for non workspace builds
+                      (forEachComponent (component: ''
+                      mkdir -p "${component}"
+                    '')) ] ++
                     map
                       (v:
                         let
                           nv = if isAttrs v then v else { name = "."; value = v; };
                         in
-                        "cp -vr ${nv.value} ${nv.name}"
+                        "cp -vr \"${nv.value}\" \"${nv.name}\""
```

Comment:
The quotes here could go in a separate commit. I think they are the only change that would be just a simple improvement to the existing code. The quotes are necessary now, because the component names are now part of the paths here, and are user-supplied.